File: Data Structures & Algorithms/is-anagram/submission-6.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        n = len(s)
        m = len(t)
        
        if n != m:
            return False
        
        seen = {}

        for i in range(n):
            if s[i] in seen:
                seen[s[i]] += 1
            else:
                seen[s[i]] = 1
        
        for i in range(n):
            if t[i] not in seen or seen[t[i]] == 0:
                return False
            seen[t[i]] -= 1
        
        return True

File: Data Structures & Algorithms/contains-duplicate-ii/submission-6.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_set<int> seen;
        for(int i =0;i<nums.size();i++){
            int j = i+1;
            while(j<nums.size() && abs(j-i) <=k){
                if(nums[j] == nums[i]){
                    return true;
                }
                j++;
            }

            seen.insert(nums[i]);

            if(seen.size() > k){
                seen.erase(nums[j-i]);
            }
        }
        return false;
    }
};